docs/mkdocs.yml

```
@@ -0,0 +1,85 @@
+
+# Navigation Structure
+nav:
+  - General:                    # General project information
+      - Home: README.md                         # Main landing page
+      - Installation: docs/tutorials/install.md # Installation guide
+  - Documentation:              # Main documentation section
+      - Source: src                             # Link to the source directory
+      - toolchains                              # Placeholder for toolchain-related documentation
+      - Simulations: simulations                # Documentation on simulations
+      - Tests: tests                            # Testing documentation
+  - Tutorials: docs/tutorials   # Section for tutorials
+
+  - MkDocs: docs/README.md      # Documentation on how to use MkDocs for this project
+
+  - API:  # API reference documentation
+      - Gyselalib++:
+          - 'Classes':  # Class-related documentation
+              - 'Class List': 'gyselalibxx/annotated.md'
+              - 'Class Index': 'gyselalibxx/classes.md'
+              - 'Class Hierarchy': 'gyselalibxx/hierarchy.md'
+          - 'Namespaces':  # Namespace-related documentation
+              - 'Namespace List': 'gyselalibxx/namespaces.md'
+              - 'Namespace Members': 'gyselalibxx/namespace_members.md'
+              - 'Namespace Member Functions': 'gyselalibxx/namespace_member_functions.md'
+              - 'Namespace Member Variables': 'gyselalibxx/namespace_member_variables.md'
+              - 'Namespace Member Typedefs': 'gyselalibxx/namespace_member_typedefs.md'
+              - 'Namespace Member Enumerations': 'gyselalibxx/namespace_member_enums.md'
+          - 'Functions': 'gyselalibxx/functions.md'  # Function documentation
+          - 'Variables': 'gyselalibxx/variables.md'  # Global variables
+          - 'Macros': 'gyselalibxx/macros.md'  # Macro definitions
+          - 'Files': 'gyselalibxx/files.md'  # File-level documentation
+
+# Theme Configuration
+theme: readthedocs  # Uses the ReadTheDocs theme for the site
+
+# Plugins for Extended Functionality
+plugins:
+  - mkdocs-pdf                            # Enables PDF export of documentation
+  - search                                # Provides search functionality
+  - include_dir_to_nav:                   # Automatically includes directories in navigation
+      flat: false                         # Preserves directory structure
+      file_name_as_title: true            # Uses file names as titles in navigation
+  - mkdoxy:                               # Integrates Doxygen-generated documentation into MkDocs
+      projects:
+        gyselalibxx:
+          src-dirs: ../src                # Source directory containing C++ code
+          full-doc: True                  # Generates full documentation
+          doxy-cfg:                       # Doxygen configuration settings
+            FILE_PATTERNS: "*.cpp *.h*"   # Files to process
```

Do you mind if we change this to:
```suggestion
            FILE_PATTERNS: "*.cpp *.hpp"   # Files to process
```
?
We don't use `*.h` and `*.h*` is catching my `.*.hpp.swp` files (backup files created by vim while I have the file open) and causing crashes

---

ahh  .hpp is good!